Add ability to recognize links to the same page (#16994)
nvaccess/nvda: user_docs/en/changes.md

* When editing in Microsoft PowerPoint text boxes, you can now move per sentence with `alt+upArrow`/`alt+downArrow`. (#17015, @LeonarddeR)
* In Mozilla Firefox, NVDA will report the highlighted text when a URL containing a text fragment is visited. (#16910, @jcsteh)

> * NVDA can report when a link destination points to the current page (#141, @LeonarddeR, @nvdaes)

Reviewer: ```suggestion
* NVDA can now report when a link destination points to the current page. (#141, @LeonarddeR, @nvdaes)
```

Add-ons will need to be re-tested and have their manifest updated.
* Updated Comtypes to 1.4.6. (#17061, @LeonarddeR)
* `ui.browseableMessage` may now be called with options to present a button for copying to clipboard, and/or a button for closing the window. (#17018, @XLTechie)
* Several additions to identify link types (#16994, @LeonarddeR, @nvdaes)

>   * A new `utils/urlUtils` module with different functions to determine link types

Reviewer: ```suggestion
  * A new `utils.urlUtils` module with different functions to determine link types
```

  * A new `INTERNAL_LINK` state has been added to `controlTypes.states.State`

>   * A new `linkType` property has been added on `NVDAObject`. It queries the `treeInterceptor` by default, if any

Reviewer: ```suggestion
  * A new `linkType` property has been added on `NVDAObject`.
  It queries the `treeInterceptor` by default, if any.
```

  * `BrowseModeTreeInterceptor` object has a new `documentUrl` property
  * `BrowseModeTreeInterceptor` object has a new `getLinkTypeInDocument` method which accepts an URL to check the link type of the object

>   * A `toggleBooleanValue` helper function has been added to `globalCommands`. It can be used in scripts to report the result when a boolean is toggled in `config.conf`

Reviewer: ```suggestion
  * A `toggleBooleanValue` helper function has been added to `globalCommands`.
  It can be used in scripts to report the result when a boolean is toggled in `config.conf`
```

* It is now possible to redirect objects retrieved from on-screen coordinates, by using the `NVDAObject.objectFromPointRedirect` method. (#16788, @Emil-18)
* Running SCons with the parameter `--all-cores` will automatically pick the maximum number of available CPU cores. (#16943, #16868, @LeonarddeR)
* Developer info now includes information on app architecture (such as AMD64) for the navigator object. (#16488, @josephsl)

> * In the gecko_ia2 virtual buffer backend, the accValue is exposed for links (#141, @LeonarddeR).

Reviewer 2: Thanks for mentioning me here, but I don't think this really needs a changelog entry.
```suggestion
```

* Anew utils/urlUtils/isSamePageUrl function has been added to check links (#141, @LeonarddeR)
* A new INTERNAL_LINK state has been added to controlTypes.states.State (#141, @nvdaes)

> * A new isInternalLink property has been added for NVDAObjects.IAccessible.ia2Web.Ia2Web objects (#141, @nvdaes)

Reviewer 2: ```suggestion
* A new isInternalLink property has been added on `NVDAObject`. It queries the treeInterceptor by default, if any. (#141, @nvdaes)
```

* The comment command in Microsoft Word and notes command in Microsoft Excel can now be pressed twice to show the comment or note in a browsable message. (#16800, #16878, @Cary-Rowen)
* NVDA can now be configured to report font attributes in speech and braille separately. (#16755)
* The timeout to perform a multiple keypress is now configurable; this may be especially useful for people with dexterity impairment. (#11929, @CyrilleB79)
* When performing a braille cursor routing action, NVDA can now automatically speak the character at the cursor. (#8072, @LeonarddeR)

>   * This option is disabled by default. You can enable "Speak character when routing cursor in text" in NVDA's braille settings.

Reviewer: should this be here?

Author: No, fixed now

* It is now possible to redirect objects retrieved from on-screen coordinates, by using the `NVDAObject.objectFromPointRedirect` method. (#16788, @Emil-18)
* Running SCons with the parameter `--all-cores` will automatically pick the maximum number of available CPU cores. (#16943, #16868, @LeonarddeR)
* Developer info now includes information on app architecture (such as AMD64) for the navigator object. (#16488, @josephsl)
* A new utils/urlUtils with different functions to determine link types has been added to check links (#141, @LeonarddeR, @nvdaes)
* A new INTERNAL_LINK state has been added to controlTypes.states.State (#141, @nvdaes)
* A new linkType property has been added on `NVDAObject`. It queries the treeInterceptor by default, if any. (#141, @LeonarddeR, @nvdaes)
* BrowseModeTreeInterceptor object has a new documentUrl property (#141, @LeonarddeR)

> * BrowseModeTreeInterceptor object has a new getLinkTypeInDocument method which accepts an URL to check the link type of the object (#141, @LeonarddeR, @nvdaes)

Reviewer: can you please group these items and add markdown code formatting around code symbols. The reference should be the PR reference so people can view the code changes directly
